src/UI/Component/Modal/Factory.php

```diff
@@ -0,0 +1,153 @@
+	 *
+	 * @return \ILIAS\UI\Component\Modal\Interruptive
+	 */
+	public function interruptive($title, $message, $form_action);
+
+
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *     Interruptive items are displayed in an interruptive modal and represent the object(s) being affected
+	 *     by the critical action, e.g. deleting.
+	 *   composition:
+	 *   effect:
+	 * rules:
+	 * ---
+	 *
+	 * @param string $id
+	 * @param string $title
+	 * @param string $description
+	 *
+	 * @return InterruptiveItem
+	 */
+	public function interruptiveItem($id, $title, $description = '');
+
+
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *     Round-Trip Modals are to be used if the context would be lost by performing this action otherwise.
+	 *     Round-Trip Modals accommodate sub-workflows within an overriding workflow. The Round-Trip Modal ensures that the users does
+	 *     not leave the trajectory of the overriding workflow. This is typically the case if an ILIAS service is called for while
+	 *     in working in an object.
+	 *   composition: >
+	 *     Round-Trip Modals are completed by a well defined sequence of only a few steps that might be displayed on a sequence of different
+	 *     modals connected through some “next” Button.
+	 *   effect: >
+	 *     Round-Trip Modals perform sub-workflow involving some kind of user input. Sub-workflow is completed and user is returned
+	 *     to starting point allowing for continuing the overriding workflow.
+	 *
+	 * rules:
+	 *   usage:
+	 *     1: >
+	 *       Round-Trip Modals MUST contain at least two buttons at the bottom of the modals: a button to cancel (right) the workflow
+	 *       and a button to finish or reach the next step in the workflow (left).
+	 *     2: >
+	 *       Round-Trip Modals SHOULD be used, if the user would lose the context otherwise. If the action can be performed within
+	 *       the same context (e.g. add a post in a forum, edit a wiki page), a Round-Trip Modal MUST NOT be used.
+	 *     3: >
+	 *       When the workflow is completed, Round-Trip Modals SHOULD show the same view as was displayed when initiating the Modal.
+	 *     4: >
+	 *       Round-Trip Modals SHOULD NOT be used to add new items of any kind since adding item is a linear workflow redirecting to the
+	 *       newly added item setting- or content-tab.
+	 *     5: >
+	 *       Round-Trip Modals SHOULD NOT be used to perform complex workflows.
+	 *
+	 * ---
+	 * @param string                                    $title
+	 * @param Component\Component|Component\Component[] $content
+	 *
+	 * @return \ILIAS\UI\Component\Modal\RoundTrip
+	 */
+	public function roundtrip($title, $content);
+
+
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *     The Lightbox Modal displays a full of data such as documents, images or a movie inside a Lightbox.
+	 *   composition: >
+	 *     If multiple pages or images are to be displayed in one Lightbox Modal, they can flipped through.
```

Comment:
@Amstutz @klees I'm really not sure what I should write in composition now - can you give me a copy paste sentence please?